src/components/views/messages/TextualBody.js

```diff
@@ -143,9 +143,15 @@ module.exports = React.createClass({
         if (this.props.showUrlPreview && !this.state.links.length) {
             var links = this.findLinks(this.refs.content.children);
             if (links.length) {
-                this.setState({ links: links.map((link)=>{
-                    return link.getAttribute("href");
-                })});
+                // de-dup the links
+                const seen = new Set();
+                links = links.filter((link) => {
```

Comment:
> Is this not the same as links = Array.from(new Set(links));

no, because afaik that will drop the ordering of the list.

>  links = links.map((link) => link.getAttribute("href"));

I moved that into `findLinks`.